trestle/core/commands/author/catalog.py

```diff
@@ -174,55 +175,75 @@ def assemble_catalog(
 
         Notes:
             If the destination catalog_name model already exists in the trestle project, it is overwritten.
-            If no original catalog name is provided, the catalog is created anew using only the markdown content.
+            If a parent catalog is not specified, the assembled catalog will be used as the parent if it exists.
+            If no parent catalog name is available, the catalog is created anew using only the markdown content.
         """
         md_dir = trestle_root / md_name
         if not md_dir.exists():
             raise TrestleError(f'Markdown directory {md_name} does not exist.')
 
+        # assemble the markdown controls into md_catalog
         md_catalog_interface = CatalogInterface()
         try:
             md_catalog = md_catalog_interface.read_catalog_from_markdown(md_dir, set_parameters)
         except Exception as e:
             raise TrestleError(f'Error reading catalog from markdown {md_dir}: {e}')
         if md_catalog_interface.get_count_of_controls_in_catalog(True) == 0:
-            raise TrestleError(f'No controls were loaded from markdown {md_dir}.  No catalog.json created.')
+            raise TrestleError(f'No controls were loaded from markdown {md_dir}.  No catalog created.')
 
-        # if original catalog given then merge the markdown controls into it
-        if orig_cat_name:
+        # this is None if it doesn't exist yet
+        assem_cat_path = ModelUtils.full_path_for_top_level_model(trestle_root, assem_cat_name, Catalog)
+
+        # if original cat is not specified, use the assembled cat but only if it already exists
+        if not parent_cat_name and assem_cat_path:
+            parent_cat_name = assem_cat_name
+
+        # default to JSON but allow override later if other file type found
+        new_content_type = FileContentType.JSON
+
+        # if we have parent catalog then merge the markdown controls into it
+        # the parent can be a separate catalog or the destination assembled catalog if it exists
+        # but this is the catalog that the markdown is merged into in memory
+        if parent_cat_name:
             try:
-                orig_cat, _ = ModelUtils.load_top_level_model(trestle_root, orig_cat_name, Catalog)
+                parent_cat, parent_cat_path = ModelUtils.load_top_level_model(trestle_root, parent_cat_name, Catalog)
             except Exception as e:
```

Comment:
Let the original exception fall up the stack, or at least catch specific exception